### PR TITLE
Adding Pair class to get rid of javafx dependency

### PR DESCRIPTION
**Pull request by cgtz**
javafx does not seem to be included in many versions of java 7, so we
decided to make our own Pair class.

*Reviewers:* @vgkholla, @pnarayanan 
*Time:* 10 min.

### CODE REVIEW

File: ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    assertEquals("Invalid first item.", first, pairOne.getFirst());
+    assertEquals("Invalid second item.", second, pairOne.getSecond());
+    assertEquals("These pairs should be equivalent", pairTwo, pairOne);
+    assertFalse("pairOne and pairThree should not be equal", pairOne.equals(pairThree));
+    assertFalse("pairOne and pairFour should not be equal", pairOne.equals(pairFour));
+    assertEquals("Hashcodes should be the same", pairOne.hashCode(), pairTwo.hashCode());
+    assertEquals("Pair{first=" + first + ", second=" + second + "}", pairOne.toString());
+  }
+}
```
Comment (vgkholla) [inline ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java]: new line required

File: ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    Pair<String, Long> pairOne = new Pair<>(first, second);
+    Pair<String, Long> pairTwo = new Pair<>(first, second);
+    Pair<String, Long> pairThree = new Pair<>(first, second + 1);
+    Pair<Long, String> pairFour = new Pair<>(second, first);
+
+    assertEquals("Invalid first item.", first, pairOne.getFirst());
+    assertEquals("Invalid second item.", second, pairOne.getSecond());
+    assertEquals("These pairs should be equivalent", pairTwo, pairOne);
+    assertFalse("pairOne and pairThree should not be equal", pairOne.equals(pairThree));
+    assertFalse("pairOne and pairFour should not be equal", pairOne.equals(pairFour));
```
Comment (vgkholla) [inline ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java]: also make sure hashCodes are unequal?

File: ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+public class PairTest {
+  @Test
+  public void testPair() {
+    String first = "abcdef";
+    Long second = 23L;
+
+    Pair<String, Long> pairOne = new Pair<>(first, second);
+    Pair<String, Long> pairTwo = new Pair<>(first, second);
+    Pair<String, Long> pairThree = new Pair<>(first, second + 1);
```
Comment (vgkholla) [inline ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java]: need one where the second is equal but the first isn't ?

File: ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+package com.github.ambry.utils;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+public class PairTest {
```
Comment (vgkholla) [inline ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java]: some token documentation for the class and test.

Comment (cgtz) [inline ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java]: added docs

File: ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java (hunk 3 above)
Comment (cgtz) [inline ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java]: added this test cases

File: ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java (hunk 2 above)
Comment (cgtz) [inline ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java]: added unequal hashcode tests.

File: ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java (hunk 1 above)
Comment (cgtz) [inline ambry-utils/src/test/java/com.github.ambry.utils/PairTest.java]: added newlines

**Comment by cgtz** [comment]
Builds on java 7 and 8

**Comment by pnarayanan** [comment]
LGTM

**Pull request merged**